app/components/candidate_interface/application_dashboard_course_choices_component.rb

```diff
@@ -109,13 +109,13 @@ def interview_row(application_choice)
 
     def conditions_row(application_choice)
       return unless (application_choice.pending_conditions? || application_choice.offer?) &&
-        (application_choice.offer.present? && !application_choice.offer['conditions'].count.zero?)
+        (application_choice.offer.present? && application_choice.offer.conditions.any?)
 
       {
-        key: 'Condition'.pluralize(application_choice.offer['conditions'].count),
+        key: 'Condition'.pluralize(application_choice.offer.conditions.count),
         value: render(
           OfferConditionsReviewComponent.new(
-            conditions: application_choice.offer['conditions'],
+            conditions: application_choice.offer.conditions.map(&:text),
```

Comment:
might be nice to have a method in `offer` that does `conditions.map(&:text)` and we can invoke everywhere?